File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/LeftKeyCommandPressUp.java
package com.example.try_gameengine.remotecontroller;

import android.view.MotionEvent;

import com.example.try_gameengine.remotecontroller.RemoteController.CommandType;

/**
 * left key press up.
 * @author irons
 *
 */
public class LeftKeyCommandPressUp implements Command{
	LeftKey leftKey;
	private int pointerId = -1;
	
	public LeftKeyCommandPressUp(LeftKey leftKey){
		this.leftKey = leftKey;
	}
	
	@Override
	public CommandType execute() {
		return CommandType.LeftKeyUpCommand;
	}

	@Override
	public boolean checkExecute(float x, float y, MotionEvent event) {
		return leftKey.pressUp(x, y, event);
	}

	@Override
	public void setMotionEventPointerId(int pointerId) {
		this.pointerId = pointerId;
	}

	@Override
	public int getMotionEventPointerId() {
		return pointerId;
	}

}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/LeftKeyCommandPressUp.kt
package com.example.try_gameengine.remotecontroller

import android.view.MotionEvent

/**
 * left key press up.
 * @author irons
 // */
class LeftKeyCommandPressUp(leftKey: LeftKey) : Command {
    var leftKey: LeftKey
    private var pointerId = -1

    init {
        this.leftKey = leftKey
    }

    override fun execute(): RemoteController.CommandType {
        return RemoteController.CommandType.LeftKeyUpCommand
    }

    override fun checkExecute(x: Float, y: Float, event: MotionEvent?): Boolean {
        return leftKey.pressUp(x, y, event)
    }

    override fun setMotionEventPointerId(pointerId: Int) {
        this.pointerId = pointerId
    }

    override fun getMotionEventPointerId(): Int {
        return pointerId
    }
}
